.github/workflows/checksrc.yml

```diff
@@ -57,11 +53,15 @@ jobs:
         run: |
           sudo rm -f /etc/apt/sources.list.d/microsoft-prod.list
           sudo apt-get update
-          sudo apt-get install codespell
+          sudo apt-get install codespell python3-pip
+          python3 -m pip install cmakelint
 
       - name: spellcheck
         run: codespell --skip src/tool_hugehelp.c -I .github/scripts/codespell-ignore.txt include src lib
 
+      - name: cmakelint
+        run: find . -type f \( -name CMake\* -o -name \*.cmake \) \! -name \*.h.cmake -print0 | xargs -0 cmakelint --spaces=2 --linelength=132 || echo Ignoring errors
```

Comment:
What do you think of moving out this line to a script and using `git ls-files` as a filelist source? It would allow running the script locally (in a dirty sandbox).

Also, these options silence all three issues detected:
```
--filter=-whitespace/indent,-convention/filename,-package/stdargs
```

- `whitespace/indent` has bogus warnings, and it also failed to catch obvious indentation errors, so we're probably fine without it till fixed.
- `convention/filename`, `-package/stdargs` both insist on using an uppercase package name. As discussed earlier, this is an option, but nobody seems to be following this convention. After thinking about it, I went with the mixed-case option. Unless we override this and rename most Find modules to uppercase, these warnings are not useful.

For these reasons I think it's safe to suppress them, and possibly even allow `cmakelint` to fail for other cases. E.g. to pick up any overly long lines.

Reply:
I wasn't sure how many of those issues you were planning on fixing and how many were just limitations in the tool. I've added the filter you suggested and created a script so it can be run locally, but using find instead of git to get the file list so that it will also work from tarballs.

Reply:
Thanks! The find method picks up things non-repo, too (entirely my faul, but there is a lot of non-compliand stuff in there). I'd imagine this script is primarily useful for people who plan to push code to the Git repo, so assuming a sandbox (and thus `git ls-files` should be fine?

Reply:
I'd be surprised if I ran cmakelint.sh on my code and it silently skipped checking my cmakefiles just because they weren't committed yet. I'd want it to check everything, regardless of their commit status and regardless of whether or not I was working in a private repository that wasn't git. The fact that it finds non-compliant stuff in local files sounds like a feature to me ☺

Reply:
Maybe, but adding new CMake files happens very rarely, and checking for lint issues before committing the first iteration is a tiny fraction in the life of a CMake source file. Before pushing, it's already committed and appears in `git ls-files`, ready to be checked.

`find .` also picks up generated CMake files for all out-of-tree build subdirs :( I almost always have some of those while working on patches.

How about this version which works both in non-Git and Git:
```shell
# Run cmakelint on the curl source code. cmakelint can be installed from PyPi
# with the command "python3 -m pip install cmakelint".
{
  if git rev-parse --is-inside-work-tree >/dev/null 2>&1; then
    git ls-files
  else
    find . -type f
  fi
} | grep -E '(/CMake|\.cmake$)' | grep -v -E '(\.h\.cmake|\.in)$' | xargs \
cmakelint \
  --spaces=2 --linelength=132 \
  --filter=-whitespace/indent,-convention/filename,-package/stdargs \
  "$@"
```

Reply:
As discussed, the script now takes any filename arguments first, then falls back to git then find, similar to your suggestion.

Reply:
Nice! Thank you, Dan!